cibuildwheel/architecture.py

```diff
@@ -93,8 +93,12 @@ def auto_archs(platform: PlatformName) -> set[Architecture]:
             # x86_64 machines can run i686 containers
             result.add(Architecture.i686)
 
-        if platform == "windows" and Architecture.AMD64 in result:
-            result.add(Architecture.x86)
+        if platform == "windows":
+            if Architecture.ARM64 in result:
+                result.add(Architecture.AMD64)
+                result.add(Architecture.x86)
```

Comment:
I was torn. I think the difference here from macOS is that Windows isn't deprecating the old architectures - they'll all continue to exist. An ARM64 Windows can build and test all architectures, while others can't test the ARM64 ones. So it becomes the one-stop shop for doing them all.

But in continuing to write this reply, I've convinced myself that I agree with you :) The two options for a CI config would be:
* run all Windows builds on x64 and set env variable to also build (but not test) for ARM64
* run x86/x64 builds and tests on x64 (default settings) and ARM64 on native (also default setting)

Reply:
Fair, though in practice it's only harder on Windows because people assume that cross-compilation doesn't exist (e.g. using `platform.machine()` to determine the target architecture). The compiler toolchain is configured and used identically to native compilation, and people have been using cross-compilation (x86<->x64 or vice versa) for years already.

It's not at all like the Linux/macOS cross-compilation process, though in fairness that's because the compilation process is entirely different to begin with. But cross-compile is much closer to normal compile on Windows than it would be on the others.